docs/api.rst

```diff
@@ -1780,6 +1784,45 @@ Deletions also support :code:`Prefer: return=representation` plus :ref:`v_filter
 
   Beware of accidentally deleting all rows in a table. To learn to prevent that see :ref:`block_fulltable`.
 
+Limited Updates/Deletions
+=========================
+
+You can limit the amount of affected rows by :ref:`update` or :ref:`delete` with the ``limit`` query parameter.
+
+.. tabs::
+
+  .. code-tab:: http
+
+    PATCH /users?limit=10&last_login=lt.2017-01-01 HTTP/1.1
+
+    { "status": "inactive" }
+
+  .. code-tab:: bash Curl
+
+    curl -X PATCH "/users?limit=10&last_login=lt.2020-01-01" \
+      -H "Content-Type: application/json" \
+      -d '{ "status": "inactive" }'
+
+.. tabs::
+
+  .. code-tab:: http
+
+    DELETE /users?limit=10&status=eq.inactive HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl -X DELETE "http://localhost:3000/users?limit=10&status=eq.inactive"
+
+Using ``offset`` to target a different subset of rows is also possible.
+
+.. note::
+
+  This feature obeys the following rules:
+
+   - If the table has a PK(and it's available on the :ref:`schema_cache`), it will use the PK to order and limit the rows. This requires a SELECT privilege on the PK columns.
+   - If no PK is available, it will fallback to using the table's `ctid <https://www.postgresql.org/docs/current/ddl-system-columns.html>`_ for the same purpose. This requires a SELECT privilege on the ``ctid``.
+   - It will work on a view if the PK or `ctid <https://www.postgresql.org/docs/current/ddl-system-columns.html>`_ is included as one of its columns.
+
```

Comment:
```suggestion
  This will only work, if at least one of the following applies:

   - The table has a PK, which can be used to limit the rows. This requires a SELECT privilege on the PK columns.
   - The target is a table, in which case the `ctid <https://www.postgresql.org/docs/current/ddl-system-columns.html>`_ column can be used for the same purpose. This requires a SELECT privilege on ``ctid``.
   - The target is a view, which includes either the table's PK or `ctid <https://www.postgresql.org/docs/current/ddl-system-columns.html>`_ column as one of its columns.
   
```

I have two questions at this point:
- Is granting column level select privileges on the ctid column possible? Or is this something to have by default?
- The view-thing reads like it could be a challenge to implement right. Will take a careful look in the implementation PR.

Reply:
> Is granting column level select privileges on the ctid column possible?

Yes, it's possible but a little weird since it's a hidden column: `GRANT SELECT(ctid) ON TABLE tbl`. That's why I was thinking in the implementation to leave the `ctid` option out but it's just too good that it works on all tables.

Of course a SELECT on all columns would work as well.

Reply:
> The view-thing reads like it could be a challenge to implement right

As long as the PK is detected on the view it should be fine, this is probably the most common case.

Reply:
This is now updated to match https://github.com/PostgREST/postgrest/pull/2211